lib/resque/worker.rb

```diff
@@ -651,7 +651,7 @@ def unregister_worker(exception = nil)
       message = exception_while_unregistering.message
       if exception
         message = message + "\nOriginal Exception (#{exception.class}): #{exception.message}\n" +
-                            "  #{exception.backtrace.join("  \n")}"
+                            "  #{(exception.backtrace || []).join("  \n")}"
```

Comment:
The `PruneDeadWorkerDirtyExit` exception does not since it's never thown, only created at https://github.com/resque/resque/blob/master/lib/resque/worker.rb#L578